integration/log_integration_test.go

```diff
@@ -276,6 +293,55 @@ func checkLogRootHashMatches(logID int64, tree *merkle.InMemoryMerkleTree, clien
 	return nil
 }
 
+// checkInclusionProofsAtIndex obtains and checks proofs at tree sizes from zero up to 2 x the sequencing
+// batch size (or number of leaves queued if less). The log should only serve proofs for indices in a tree
+// at least as big as the index where STHs where the index is a multiple of the sequencer batch size. All
+// proofs returned should match ones computed by the alternate Merkle Tree implementation, which differs
+// from what the log uses.
+func checkInclusionProofsAtIndex(testIndex int64, logID int64, tree *merkle.InMemoryMerkleTree, client trillian.TrillianLogClient, params testParameters) error {
+	for treeSize := int64(0); treeSize < min(params.leafCount, int64(2 * params.sequencerBatchSize)); treeSize++ {
+		ctx, cancelFunc := getRPCDeadlineContext()
+		resp, err := client.GetInclusionProof(ctx, &trillian.GetInclusionProofRequest{LogId: logID, LeafIndex:testIndex, TreeSize:int64(treeSize)})
+		cancelFunc()
+
+		if testIndex >= treeSize {
+			// Firstly, if the index is larger than the tree size we cannot have a valid proof
+			if err == nil {
+				return fmt.Errorf("log returned proof for index: %d, tree is only size %d", testIndex, treeSize)
+			}
+		} else {
+			// Secondly, if we're not at a valid STH tree size then we can't have a proof
+			if treeSize == 0 || (treeSize % int64(params.sequencerBatchSize)) != 0 {
```

Comment:
Hmmm. Yes it would be allowed to but if the current implementation returns something for these requests then it's broken.